data-prepper-plugins/obfuscate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/obfuscation/action/OneWayHashAction.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.obfuscation.action;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.processor.obfuscation.ObfuscationProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@DataPrepperPlugin(name = "hash", pluginType = ObfuscationAction.class, pluginConfigurationType = OneWayHashActionConfig.class)
+public class OneWayHashAction implements ObfuscationAction {
+    
+
+    private MessageDigest messageDigest = null;
+    private byte[] salt;
+    private String saltKey;
+    private static final Logger LOG = LoggerFactory.getLogger(ObfuscationProcessor.class);
+
+    @DataPrepperPluginConstructor
+    public OneWayHashAction(final OneWayHashActionConfig config) {
+        
+        this.salt = config.getSalt();
+        this.saltKey = config.getSaltKey();
+        try{
+            messageDigest = MessageDigest.getInstance(config.getHashFormat());            
+        } catch (NoSuchAlgorithmException noSuchAlgorithmException){
+            LOG.error("There was an exception while processing Event [{}]", noSuchAlgorithmException);
```

Comment:
Since you are catching this exception, you know the problem is there is no such algorithm. So we can improve the error. 

Also, we should re-throw because this will cause the pipeline to fail to run. As it is now, the pipeline would start, all processing would fail because you'd get a NullPointerException when trying to use the `messageDigest`.

```
LOG.error("The hash format provided ({}) is not a known algorithm [{}]", config.getHashFormat(), noSuchAlgorithmException);
throw new RuntimeException(ex);
```

Reply:
Addressed.